source/common/network/dns_resolver/dns_factory_util.cc

```diff
@@ -81,7 +81,9 @@ Network::DnsResolverFactory& createDnsResolverFactoryFromTypedConfig(
   ENVOY_LOG_MISC(debug, "create DNS resolver type: {}", typed_dns_resolver_config.name());
   auto& factory =
       Config::Utility::getAndCheckFactory<Network::DnsResolverFactory>(typed_dns_resolver_config);
-  factory.initialize();
+  if (factory.name() == std::string(CaresDnsResolver)) {
```

Comment:
I think it'd be cleaner if unconditionally call `initialize()`, and unconditionally call `terminate()` also. They do have empty-impls in base classes so it'll work fine and be much cleaner.

If some other DNS resolver wants initialize/terminate semantics they can just override the default and not add special cases.

Reply:
+1 I was going to say the same thing. Please just do it by category.

Reply:
Done.